tools/tsp-client/src/npm.ts

```diff
@@ -0,0 +1,51 @@
+import * as path from "node:path";
+import { readFile, writeFile } from "node:fs/promises";
+import { spawn } from "node:child_process";
+import { fileURLToPath } from "node:url";
+
+export async function createPackageJson(rootPath: string, deps: Set<string>): Promise<void> {
+  const dependencies: Record<string, string> = {};
+
+  for (const dep of deps) {
+    dependencies[dep] = "latest";
+  }
+
+  const packageJson = JSON.stringify({
+    dependencies,
+  });
+
+  const filePath = path.join(rootPath, "package.json");
+  await writeFile(filePath, packageJson);
+}
+
+export function installDependencies(workingDir: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const npm = spawn("npm", ["install", "--no-lock-file"], {
```

Comment:
I suspect this was mimicing the existing scripts. Note there is a PR in flight for supporting of the lock file, see https://github.com/Azure/azure-sdk-tools/pull/6823.

Reply:
This is mostly because I'm dynamically generating the lockfile as outside of JS the other repo lockfiles only appear to pull in the emitter package rather than all the dependencies the TSP has. 

It'd still be useful to parse the local lockfile in the repo, but I figured that could be added in.

Reply:
Tracking issue to use lock file: https://github.com/Azure/azure-sdk-tools/issues/7057